### PR TITLE
Implement 3 ULDUM cards

### PR DESCRIPTION
**Pull request by doobee98**
This revision inculdes:

- Implement 3 ULDUM cards
    - Sahket Sapper (ULD_280)
    - Hooked Scimitar (ULD_285)
    - Shadow of Death (ULD_286)

### CODE REVIEW

File: Tests/UnitTests/PlayMode/CardSets/UldumCardsGenTests.cpp [hunk 1]
```diff
@@ -2524,6 +2524,160 @@ TEST_CASE("[Rogue : Minion] - ULD_231 : Whirlkick Master")
...
+    const auto card3 = 
+        Generic::DrawCard(opPlayer, Cards::FindCardByName("Eviscerate"));
+
+    game.Process(opPlayer, PlayCardTask::Minion(card1));
+    game.Process(opPlayer, PlayCardTask::Minion(card2));
+    game.Process(opPlayer, PlayCardTask::SpellTarget(card3, card1));
+    
+    CHECK_EQ(curField.GetCount(), 0);
+    CHECK_EQ(opField.GetCount(), 0);
+    CHECK_EQ(opHand.GetCount(), 2);     // 'The Coin' and returned 'Sahket Sapper'
```
[the remark this answers is not in the document]
Comment (doobee98) [reply Tests/UnitTests/PlayMode/CardSets/UldumCardsGenTests.cpp:2585]: You mean current player's hand? I am checking two player's field count in line 2583 and 2584.

File: Sources/Rosetta/PlayMode/CardSets/UldumCardsGen.cpp [hunk 2]
```diff
@@ -1460,6 +1474,29 @@ void UldumCardsGen::AddRogue(std::map<std::string, CardDef>& cards)
...
+    power.ClearData();
+    power.AddPowerTask(std::make_shared<CustomTask>(
+        [](Player* player, Entity* source, [[maybe_unused]] Playable* target) {
+            if (target == nullptr)
+            {
+                return;
+            }
+            
+            std::map<GameTag, int> tags;
+            tags.emplace(GameTag::TEMPORARY_SAVED_DBFID, target->card->dbfID);
```
[the remark this answers is not in the document]
Comment (doobee98) [reply Sources/Rosetta/PlayMode/CardSets/UldumCardsGen.cpp:1486]: I only check 'Swampqueen Hagatha' to implement it and couldn't find other substitutions.
Thanks for suggestion.

File: Sources/Rosetta/PlayMode/CardSets/UldumCardsGen.cpp [hunk 3]
```diff
@@ -1460,6 +1474,29 @@ void UldumCardsGen::AddRogue(std::map<std::string, CardDef>& cards)
     // - REQ_TARGET_TO_PLAY = 0
     // - REQ_MINION_TARGET = 0
     // --------------------------------------------------------
+    power.ClearData();
+    power.AddPowerTask(std::make_shared<CustomTask>(
+        [](Player* player, Entity* source, [[maybe_unused]] Playable* target) {
+            if (target == nullptr)
+            {
+                return;
+            }
```
[the remark this answers is not in the document]
Comment (doobee98) [reply Sources/Rosetta/PlayMode/CardSets/UldumCardsGen.cpp:1483]: After removing this code, 'Puzzle Box of Yogg-Saron' (ULD_216) testcases make error.
So I won't remove it for now.

File: Tests/UnitTests/PlayMode/CardSets/UldumCardsGenTests.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (utilForever) [reply Tests/UnitTests/PlayMode/CardSets/UldumCardsGenTests.cpp:2585]: Oh, sorry. Yes, current player's hand.

File: Sources/Rosetta/PlayMode/CardSets/UldumCardsGen.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (utilForever) [reply Sources/Rosetta/PlayMode/CardSets/UldumCardsGen.cpp:1483]: It was my mistake. Thank you for debugging.

File: Includes/Rosetta/Common/Enums/GameTagCustom.def [hunk 4]
```diff
@@ -7,4 +7,4 @@ X(NUM_SPELLS_PLAYED_THIS_TURN)
 X(NUM_SPELLS_PLAYED_THIS_GAME)
 X(NUM_CARDS_PLAYED_THIS_GAME_NOT_START_IN_DECK)
 X(LEFT_OR_RIGHT_MOST_CARD_IN_HAND)
-X(EXTRA_TRIGGER_SECRET)
\ No newline at end of file
+X(EXTRA_TRIGGER_SECRET)
```
Comment (utilForever) [inline Includes/Rosetta/Common/Enums/GameTagCustom.def:10]: Please delete it.

**Pull request merged**